src/briefcase/integrations/linuxdeploy.py

```diff
@@ -124,6 +132,228 @@ def patch_elf_header(self):
                 )
             else:
                 # We should only get here if the file at the AppImage patch doesn't have
-                # The original or patched value. If this is the case, the file is likely
+                # the original or patched value. If this is the case, the file is likely
                 # wrong and we should raise an exception.
-                raise CorruptToolError("linuxdeploy")
+                raise CorruptToolError(self.name)
+
+
+class LinuxDeployPluginBase(LinuxDeployBase):
+    """Base class for linuxdeploy plugins."""
+
+    install_msg = "{full_name} was not found; downloading and installing..."
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.env = {}
+
+        if not self.file_name.startswith("linuxdeploy-plugin-"):
+            raise BriefcaseCommandError(f"{self.file_name} is not a linuxdeploy plugin")
+
+    @property
+    def plugin_id(self):
+        return self.file_name.split(".")[0].split("-")[2]
```

Comment:
`plugin_id` is defined sometimes as a property and sometimes as a simple class variable. This makes `LinuxDeploy.plugins` a bit confusing, because it looks up `plugin_id` on the class, but when you go looking for its definition you might find the base class property, which can only be looked up on an instance, not a class.

I think it would be clearer to remove the `plugin_id` overrides in the GTK and Qt classes, and instead make `LinuxDeploy.plugins` set the known plugin names directly in a dict literal.

Reply:
Yeah - I went back and forth on this. The inconsistency is potentially confusing; using a literal in `LinuxDeploy.plugins` means duplication of information. You're probably right that being confusing is a bigger risk than the duplication, though.